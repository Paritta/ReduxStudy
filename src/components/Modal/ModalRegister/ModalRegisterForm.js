import React from "react";
import styled from "styled-components";
import { Field, reduxForm  } from "redux-form";
import oc from "open-color";
import PropTypes from "prop-types";

const propTypes = {
    handleSubmit: PropTypes.func,
    hideModal: PropTypes.func,
    AnimateTurn: PropTypes.func,
    AnimateDown: PropTypes.func
};

const defaultTypes = {
    handleSubmit() {},
    hideModal() {},
    AnimateTurn() {},
    AnimateDown() {},
};

const Wrapper = styled.div`
    padding: 30px 0 10px 0;
`;

const StyledField = styled(Field)`
    background-color: ${oc.gray[0]};
    width: 400px;
    height: 40px;
    font-size: 15px;
    border: 2px solid white;
    padding-left: 20px;
    margin-bottom: 10px;
`;

const StyledSelect = StyledField.extend`
    width: 426px;
    height: 47px;
`;

const IntroField = StyledField.extend`
    padding: 0 0 200px 20px;
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;


// 스터디 제목, 카테고리, 장소, 스터디 시간, 모집 명수, 스터디 소개
export class ModalRegisterForm extends React.Component {
    hideAnimate () {
        this.props.animateTurn();
        setTimeout(this.props.hideModal, 500);
        setTimeout(this.props.animateDown, 600);
    }

    render () {
        const { handleSubmit } = this.props;

        return (
            <Wrapper>
                <form onSubmit={ handleSubmit }>
                    <div>
                        <StyledField name="StudyTitle" component="input" type="text" required placeholder="스터디 이름"/>
                    </div>
                    <StyledSelect name="Category" component="select">
                        <option value="MaterialScience">신소재공학과</option>
                        <option value="Nursing">간호학과</option>
                        <option value="ITEngineering">IT융합응용공학과</option>
                        <option value="ComputerScience">컴퓨터공학과</option>
                        <option value="Chemical">화학공학과</option>
                    </StyledSelect>
                    <div>
                        <StyledField name="Location" component="input" type="text" required placeholder="장소"/>
                    </div>
                    <div>
                        <StyledField name="Time" component="input" type="text" required placeholder="시간"/>
                    </div>
                    <div>
                        <StyledField name="Number" component="input" type="text" required placeholder="모집 인원"/>
                    </div>
                    <div>
                        <IntroField name="Introduction" component="input" type="text" required placeholder="스터디 소개"/>
                    </div>
                    <Button type="submit">
                        등록
                    </Button>
                    <Button onClick={() => this.hideAnimate()}>나가기</Button>
                </form>
            </Wrapper>
        )
    }
};

ModalRegisterForm.propTypes = propTypes;
ModalRegisterForm.defaultTypes = defaultTypes;

ModalRegisterForm = reduxForm({
    form: "ModalRegisterForm"
})(ModalRegisterForm);

export default ModalRegisterForm;
