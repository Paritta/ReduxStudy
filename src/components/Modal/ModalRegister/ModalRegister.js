import React from "react";
import styled, {keyframes} from "styled-components";
import { bounceInLeft, bounceOutRight } from 'react-animations';
import ModalRegisterForm from "./ModalRegisterForm";
import PropTypes from "prop-types";
import { firebaseConnect, pathToJS } from "react-redux-firebase";
import { connect } from "react-redux";
import ReactLoading from "react-loading";

const propTypes = {
    Modal: PropTypes.object,
    hideModal: PropTypes.func,
    Animate: PropTypes.object,
    AnimateTurn: PropTypes.func,
    AnimateDown: PropTypes.func,
    firebase: PropTypes.object
};

const defaultTypes = {
    Modal: {},
    hideModal() {},
    Animate: {},
    AnimateTurn() {},
    AnimateDown() {}
};

const bounceInLeftAnimation = keyframes`${bounceInLeft}`;
const bounceOutRightAnimation = keyframes`${bounceOutRight}`;

const AnimationWrapper = styled.div`
    animation: 0.5s ${bounceInLeftAnimation};
`;

const AnimationOutWrapper = styled.div`
    animation: 0.5s ${props => props.AnimateOut ? bounceOutRightAnimation : ""};
`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 700px;
`;

const ModalHeader = styled.div`
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

const WrappedReactLoading = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    transform: scale(2);
    padding: 80px 0;
`;

export class ModalRegister extends React.Component {
    state = {
        isLoading: false,
        error: "",
    };

    submit = values => {
        const uid = this.props.firebase.auth().currentUser.uid;
        const username = this.props.profile.username;
        const display = username ? username : this.props.profile.displayName;

        const post = {
            values,
            author: uid,
            username: display
        };

        console.log(values);
        this.setState({ isLoading: true });
        this.props.firebase
            .push("posts", post)
            .then(()=> {
                this.setState({ isLoading: false });
                this.props.hideModal();
            })
            .catch((error) => {
                this.setState({ isLoading: false, error: error });
            });
    };


    render () {
        const { hideModal, animateTurn, animateDown, Animate } = this.props;

        return (
            <div>
                <Dimmed />
                <AnimationOutWrapper
                    AnimateOut={Animate.AnimateOut}
                >
                    <AnimationWrapper>
                        <Wrapper>
                            <ModalHeader>스터디 등록</ModalHeader>
                            <ModalBox>
                                {!this.state.isLoading
                                    ?
                                    <ModalRegisterForm
                                        onSubmit={this.submit}
                                        hideModal={hideModal}
                                        animateTurn={animateTurn}
                                        animateDown={animateDown}
                                    />
                                    :
                                    <WrappedReactLoading>
                                        <ReactLoading type="cylon" color="palevioletred"/>
                                    </WrappedReactLoading>
                                }
                            </ModalBox>
                        </Wrapper>
                    </AnimationWrapper>
                </AnimationOutWrapper>
            </div>
        )
    }
}

ModalRegister.propTypes = propTypes;
ModalRegister.defaultTypes = defaultTypes;

const WrappedModalRegister = firebaseConnect()(ModalRegister);

export default connect(
    ({ firebase }) => ({
        profile: pathToJS(firebase, "profile")
    })
)(WrappedModalRegister);