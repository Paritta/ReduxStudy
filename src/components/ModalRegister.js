import React from "react";
import styled, {keyframes} from "styled-components";
import { zoomIn } from 'react-animations';
import oc from "open-color";
import ModalRegisterForm from "./ModalRegisterForm";

const zoomInAnimation = keyframes`${zoomIn}`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 400px;
    
    animation: 1s ${zoomInAnimation};
`;

const ModalHeader = styled.div`
    @import url(//fonts.googleapis.com/earlyaccess/hanna.css);
    font-family: 'Hanna', fantasy;
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

const ModalRegister = () => {
    return (
        <div>
            ModalRegister
            <Dimmed />
            <Wrapper>
                <ModalHeader>스터디 등록</ModalHeader>
                <ModalBox>
                    <ModalRegisterForm />
                    <div>Body</div>
                    <div>Footer</div>
                    <Button>등록하기</Button>
                </ModalBox>
            </Wrapper>
        </div>
    )
};

export default ModalRegister;