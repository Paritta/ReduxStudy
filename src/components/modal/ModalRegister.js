import React from "react";
import styled, {keyframes} from "styled-components";
import { fadeIn } from 'react-animations';
import oc from "open-color";
import ModalRegisterForm from "./ModalRegisterForm";
import PropTypes from "prop-types";
import { firebaseConnect } from "react-redux-firebase";
import { connect } from "react-redux";

const propTypes = {
    Modal: PropTypes.object,
    hideModal: PropTypes.func,
};

const defaultTypes = {
    hideModal() {},
};

const fadeInAnimation = keyframes`${fadeIn}`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 700px;
`;

const ModalHeader = styled.div`
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

const AnimationWrapper = styled.div`
    animation: 1s ${fadeInAnimation};
`;

const Button = styled.button`
    background: palevioletred;
    color: white;
    
    font-size: 1em;
    margin: 1em;
    padding: 0.25em 1em;
    
    border: 2px solid palevioletred;
    border-radius: 3px;
    
    transition: all 0.3s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

class ModalRegister extends React.Component {
    submit = values => {
        console.log(values);
        this.props.firebase.push("Posts", { post: values });
    };

    render () {
        const { hideModal } = this.props;

        return (
            <div>
                <Dimmed />
                <Wrapper>
                    <AnimationWrapper>
                        <ModalHeader>스터디 등록</ModalHeader>
                        <ModalBox>
                            <ModalRegisterForm
                                onSubmit={this.submit}
                                hideModal={hideModal}/>
                        </ModalBox>
                    </AnimationWrapper>
                </Wrapper>
            </div>
        )
    }
};

ModalRegister.propTypes = propTypes;
ModalRegister.defaultTypes = defaultTypes;

const WrappedModalRegister = firebaseConnect()(ModalRegister);

export default connect(
    ({ firebase }) => ({})
)(WrappedModalRegister);