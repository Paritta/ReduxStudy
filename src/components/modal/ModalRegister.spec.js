import React from "react";
import ReactTestRenderer from "react-test-renderer"
import { ModalRegister, Wrapper } from "./ModalRegister";

describe("Check ModalRegister Component", () => {
   it("Component render correctly", () => {
       const tree = ReactTestRenderer.create(
       );
       expect(tree.toJSON()).toMatchSnapshot();
   });
});
