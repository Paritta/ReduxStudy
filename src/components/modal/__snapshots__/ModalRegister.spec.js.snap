// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check ModalRegister Component Component render correctly 1`] = `null`;
